### PR TITLE
Add AlwaysWrap functions for statusx

### PR DESCRIPTION
**Pull request by molon**
## Summary
- Add `AlwaysWrap` / `AlwaysWrapf` that always apply the caller's code/reason/message, even when `err` is already a Status error (unlike `Wrap` which silently returns the original)
- Add `AlwaysWrapCode` / `AlwaysWrapCodef` convenience variants that auto-derive reason from gRPC code
- Implementation uses `FromError` + `Clone` to preserve existing details (field violations, metadata, etc.) while overriding code/reason/message

## Test plan
- [x] Plain error wrapping
- [x] nil error returns OK (consistent with Wrap)
- [x] Overrides existing StatusError and gRPC status error
- [x] Preserves field violations and other details from original Status
- [x] Does not mutate original StatusError
- [x] Localized key matches new reason
- [x] Format string variants work correctly
- [x] All existing tests still pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: statusx/status.go [hunk 1]
```diff
@@ -458,3 +458,26 @@ func Wrap(err error, c codes.Code, reason, message string) *Status {
 func Wrapf(err error, c codes.Code, reason, format string, a ...any) *Status {
 	return Wrap(err, c, reason, fmt.Sprintf(format, a...))
 }
+
+// AlwaysWrap is like Wrap but always applies the given code, reason, and message,
+// even if err is already a Status error. The original error is preserved as the cause.
+func AlwaysWrap(err error, c codes.Code, reason, message string) *Status {
+	if err == nil {
+		return New(codes.OK, statusv1.ErrorReason_OK.String(), "")
+	}
```
[the remark this answers is not in the document]
Comment (molon) [reply statusx/status.go]: Fixed in aa308148. Added nil-error behavior documentation to AlwaysWrap docstring.

File: statusx/status.go [hunk 2]
```diff
@@ -458,3 +458,26 @@ func Wrap(err error, c codes.Code, reason, message string) *Status {
...
+	s, ok := FromError(err)
+	if ok {
+		s = Clone(s)
+	}
+	s.cause = errors.WithStack(err)
+	s.code = c
+	s.message = message
+	s.errorInfo.Reason = reason
+	// Immediately fix key to creation-time reason
+	s.localized = &statusv1.Localized{Key: s.Reason()}
```
[the remark this answers is not in the document]
Comment (molon) [reply statusx/status.go:489]: Fixed in aa308148. Added documentation that localized key is reset to match the new reason, and noted that WithLocalized/WithLocalizedArgs can be used on the result if custom localization is needed.

File: statusx/code.go [hunk 3]
```diff
@@ -70,3 +70,15 @@ func WrapCode(err error, code codes.Code, message string) *Status {
...
 }
+
+// AlwaysWrapCode is like WrapCode but always applies the given code and message,
+// even if err is already a Status error. The original error is preserved as the cause.
+func AlwaysWrapCode(err error, code codes.Code, message string) *Status {
+	return AlwaysWrap(err, code, ReasonFromCode(code).String(), message)
+}
+
+// AlwaysWrapCodef is like WrapCodef but always applies the given code and formatted message,
+// even if err is already a Status error. The original error is preserved as the cause.
```
[the remark this answers is not in the document]
Comment (molon) [reply statusx/code.go]: Fixed in aa308148. Added nil-error behavior documentation to both AlwaysWrapCode and AlwaysWrapCodef.

File: statusx/status.go [hunk 4]
```diff
@@ -458,3 +458,31 @@ func Wrap(err error, c codes.Code, reason, message string) *Status {
 func Wrapf(err error, c codes.Code, reason, format string, a ...any) *Status {
 	return Wrap(err, c, reason, fmt.Sprintf(format, a...))
 }
+
+// AlwaysWrap is like Wrap but always applies the given code, reason, and message,
+// even if err is already a Status error. The original error is preserved as the cause.
+// If err is nil, it returns an OK status (consistent with Wrap).
+//
```
[the remark this answers is not in the document]
Comment (molon) [reply statusx/status.go]: Fixed in 759c16e9. Reworded to "always sets" instead of "always applies" and documented the codes.OK + non-nil error invariant explicitly.

File: statusx/code.go [hunk 5]
```diff
@@ -70,3 +70,17 @@ func WrapCode(err error, code codes.Code, message string) *Status {
 func WrapCodef(err error, code codes.Code, format string, a ...any) *Status {
 	return Wrapf(err, code, ReasonFromCode(code).String(), format, a...)
 }
+
+// AlwaysWrapCode is like WrapCode but always applies the given code and message,
+// even if err is already a Status error. The original error is preserved as the cause.
+// If err is nil, it returns an OK status (consistent with AlwaysWrap).
+func AlwaysWrapCode(err error, code codes.Code, message string) *Status {
```
[the remark this answers is not in the document]
Comment (molon) [reply statusx/code.go]: Fixed in 759c16e9. Added reference to AlwaysWrap docs for the codes.OK edge case in both AlwaysWrapCode and AlwaysWrapCodef.

File: statusx/status_test.go [hunk 6]
```diff
@@ -287,6 +287,154 @@ func TestWrap(t *testing.T) {
...
+
+		assert.Equal(t, codes.Internal, wrapped.Code())
+		assert.Equal(t, "INTERNAL_ERROR", wrapped.Reason())
+		assert.Equal(t, "internal server error", wrapped.Message())
+	})
+
+	t.Run("overrides existing gRPC status error", func(t *testing.T) {
+		st, _ := status.New(codes.NotFound, "resource not found").WithDetails(&errdetails.ErrorInfo{
+			Reason: "NOT_FOUND",
+		})
```
[the remark this answers is not in the document]
Comment (molon) [reply statusx/status_test.go]: Fixed in 4eaa499d. Added `require.NoError(t, err)` for both the new test and the existing similar pattern in TestWrap.

File: statusx/status.go [hunk 7]
```diff
@@ -458,3 +458,35 @@ func Wrap(err error, c codes.Code, reason, message string) *Status {
...
+// When wrapping an existing Status, structural details (field violations, metadata, etc.)
+// are preserved via Clone, but the localized key is reset to match the new reason.
+// Use WithLocalized/WithLocalizedArgs on the result if custom localization is needed.
+func AlwaysWrap(err error, c codes.Code, reason, message string) *Status {
+	if err == nil {
+		return New(codes.OK, statusv1.ErrorReason_OK.String(), "")
+	}
+	s, ok := FromError(err)
+	if ok {
+		s = Clone(s)
```
[the remark this answers is not in the document]
Comment (molon) [reply statusx/status.go:482]: Fixed in 8907ed80. Now only clones when err is a *StatusError (the only case where FromError returns a shared pointer).

File: statusx/status_test.go [hunk 8]
```diff
@@ -287,6 +288,155 @@ func TestWrap(t *testing.T) {
...
+	})
+
+	t.Run("nil error", func(t *testing.T) {
+		wrapped := AlwaysWrap(nil, codes.Internal, "INTERNAL_ERROR", "internal server error")
+		require.NotNil(t, wrapped)
+		assert.Equal(t, codes.OK, wrapped.Code())
+		assert.Equal(t, statusv1.ErrorReason_OK.String(), wrapped.Reason())
+		assert.Equal(t, "", wrapped.Message())
+	})
+
```
[the remark this answers is not in the document]
Comment (molon) [reply statusx/status_test.go:310]: Fixed in e44c871e. Added the suggested test case for codes.OK + non-nil error edge case.

**Review by ZLValien — APPROVED** [review]
